packages/webapp/src/containers/Animals/Inventory/useAnimalOrBatchRemoval.ts

```diff
@@ -46,24 +60,90 @@ const useAnimalOrBatchRemoval = (
         });
+        selectedAnimalIds.push(id);
       } else if (kind === AnimalOrBatchKeys.BATCH) {
         animalBatchRemovalArray.push({
           id: entity_id,
           animal_removal_reason_id: Number(formData.reason),
           removal_explanation: formData.explanation,
           removal_date: timestampedDate,
         });
+        selectedBatchIds.push(id);
+      }
+    }
+
+    try {
+      if (animalRemovalArray.length) {
+        await mutations['removeAnimals'].trigger(animalRemovalArray).unwrap();
+        setSelectedInventoryIds(selectedInventoryIds.filter((i) => !selectedAnimalIds.includes(i)));
+        dispatch(enqueueSuccessSnackbar(t('ANIMALS.SUCCESS_REMOVE_ANIMALS', { ns: 'message' })));
+      }
+    } catch (err) {
+      dispatch(enqueueErrorSnackbar(t('ANIMALS.FAILED_REMOVE_ANIMALS', { ns: 'message' })));
+    }
+
+    try {
+      if (animalBatchRemovalArray.length) {
+        await mutations['removeBatches'].trigger(animalBatchRemovalArray).unwrap();
+        setSelectedInventoryIds(selectedInventoryIds.filter((i) => !selectedBatchIds.includes(i)));
+        dispatch(enqueueSuccessSnackbar(t('ANIMALS.SUCCESS_REMOVE_BATCHES', { ns: 'message' })));
+      }
+    } catch (err) {
+      dispatch(enqueueErrorSnackbar(t('ANIMALS.FAILED_REMOVE_BATCHES', { ns: 'message' })));
+    }
+
+    setRemovalModalOpen(false);
+  };
+
+  const handleAnimalOrBatchDeletion = async () => {
+    const animalIds: number[] = [];
+    const selectedAnimalIds: string[] = [];
+    const animalBatchIds: number[] = [];
+    const selectedBatchIds: string[] = [];
+
+    for (const id of selectedInventoryIds) {
+      const { kind, id: entity_id } = parseInventoryId(id);
+      if (kind === 'ANIMAL') {
+        animalIds.push(entity_id);
+        selectedAnimalIds.push(id);
+      } else if (kind === 'BATCH') {
+        animalBatchIds.push(entity_id);
+        selectedBatchIds.push(id);
+      }
+    }
+
+    try {
+      if (animalIds.length) {
+        await mutations['deleteAnimals'].trigger(animalIds).unwrap();
+        setSelectedInventoryIds(selectedInventoryIds.filter((i) => !selectedAnimalIds.includes(i)));
+        dispatch(enqueueSuccessSnackbar(t('ANIMALS.SUCCESS_REMOVE_ANIMALS', { ns: 'message' })));
       }
+    } catch (err) {
+      dispatch(enqueueErrorSnackbar(t('ANIMALS.FAILED_REMOVE_ANIMALS', { ns: 'message' })));
     }
 
-    animalRemovalArray.length && removeAnimals(animalRemovalArray);
-    animalBatchRemovalArray.length && removeBatches(animalBatchRemovalArray);
+    try {
+      if (animalBatchIds.length) {
+        await mutations['deleteBatches'].trigger(animalBatchIds).unwrap();
+        setSelectedInventoryIds(selectedInventoryIds.filter((i) => !selectedBatchIds.includes(i)));
+        dispatch(enqueueSuccessSnackbar(t('ANIMALS.SUCCESS_REMOVE_BATCHES', { ns: 'message' })));
+      }
+    } catch (err) {
+      dispatch(enqueueErrorSnackbar(t('ANIMALS.FAILED_REMOVE_BATCHES', { ns: 'message' })));
+    }
```

Comment:
Since you've already grouped the mutations states in the `useMutations` hook (nice!), it'd be great to take advantage of that here to already start showing a single a single snackbar. I think the only change needed would be to get rid of the `dispatch` lines in this function and add a `useEffect` within this hook that reacts to changes in `inError` and `isSuccess` returned by `useMutations`

```js
 const { mutations, isError, isSuccess } = useMutations([
    { label: 'deleteAnimals', hook: useDeleteAnimalsMutation },
    { label: 'deleteBatches', hook: useDeleteAnimalBatchesMutation },
  ]);
useEffect(() => {
    if (isSuccess) {
      dispatch(enqueueSuccessSnackbar(t('ANIMALS.SUCCESS_REMOVE_BATCHES', { ns: 'message' })));
    }
    if (isError) {
      dispatch(enqueueErrorSnackbar(t('ANIMALS.FAILED_REMOVE_BATCHES', { ns: 'message' })));
    }
  }, [isError, isSuccess]);
```

We'd still have to implement the changes in the BE to allow partial success for each individual endpoint, but this way if either of the endpoints failed we'd display the single error snackbar so we'd be halfway there.

Reply:
Ah I see! In that case yeah, I'd refactor the `useMutations` hook so that it only takes care of the "re-naming" aspect of things and doesn't infer anything from the bulk of the mutations, or maybe even get rid of the `useMutations` hook altogether? I think it might lose value once the whole awaiting for both mutations with `Promise.all` to display a single snackbar enters the picture.

Are you thinking that step of using `Promise.all` to await both mutations and show a single snackbar would be part of the "partial success" ticket's scope?